File: LudoUI/models/Board.py
from typing import Dict

from models.Tile import Tile


class Board:
    def __init__(self, id: int, game_id: int, rows:int, cols:int, tiles: Dict[str, Tile]):
        self.id = id
        self.game_id = game_id
        self.cols = cols
        self.rows = rows
        self.tiles = tiles

    def __repr__(self):
        return f"Board(id={self.id}, width={self.cols}, height={self.rows}, game_id={self.game_id}, tiles={self.tiles})"

    @classmethod
    def from_json(cls, data: Dict):
        tiles = {
            key: Tile.from_json(value)
            for key, value in data["tiles"].items()
        }
        return cls(id=data["id"], game_id=data["gameId"], rows=data["rows"], cols=data["cols"], tiles=tiles)

File: LudoAPI/Models/Board.cs
using LudoAPI.Models.Tiles;

namespace LudoAPI.Models;

public class Board
{
    public int Id { get; }
    public int GameId { get; }
    public int Rows { get; }
    public int Cols { get; }
    public Dictionary<string, Tile> Tiles { get; }

    public Board(int gameId, Dictionary<string, Tile> tiles, int rows, int cols)
    {
        Id = -1;
        GameId = gameId;
        Tiles = tiles;
        Rows = rows;
        Cols = cols;
    }

    public Board(int id, Board board)
    {
        Id = id;
        GameId = board.GameId;
        Tiles = board.Tiles;
        Rows = board.Rows;
        Cols = board.Cols;
    }
}